src/pyDataverse/api.py

```diff
@@ -185,7 +185,7 @@ def post_request(self, url, data=None, auth=False, params=None, files=None):
                 "ERROR: POST - Could not establish connection to API: {0}".format(url)
             )
 
-    def put_request(self, url, data=None, auth=False, params=None):
```

Comment:
This should stay `None`.
It is almost always better to use `None` instead of `{}` or `[]`, the common pattern being:
```python
def fun(arg=None):
    if arg is None:
        arg = {}
```

The reason is, roughly speaking, that python will process the def-line at its first pass through the code, initializing all arguments given there once and storing the references.
If the underlying argument would then, e.g., be appended to, this would modify the same object for each call. Consider:

```python
def fun(arg=[]):
    arg.append(1)
    print(arg)
fun()
fun()
fun()
```
This would print:
```
[1]
[1, 1]
[1, 1, 1]
```
However, if the expected output is [1], [1], [1], the code should be:
```python
def fun(arg=None):
    if arg is None:
        arg = []
    arg.append(1)
    print(arg)
fun()
fun()
fun()
```

While it is not so important here, I would still recommend sticking with the convention unless there's good reason not to.

For more information, see https://docs.python.org/3/faq/programming.html#why-are-default-values-shared-between-objects.